### PR TITLE
Add eth_newFilter

### LINKED ISSUE #1485: Implement `eth_newFilter`
### Problem

As a developer I'd like to be able to add a new filter, which I can later uninstall or check for changes.

Creates a filter object, based on filter options, to notify when the state changes (logs). To check if the state has changed, call `eth_getFilterChanges`. 

### Solution

Add new endpoint for creating filters which returns the id of the newly created filter.
⚠️ Filters expire after 5 minutes of inactivity (no queries).

**Parameters**
-

A transaction response object which contains the following filter information:
- fromBlock (string) - The block number as a string in hexadecimal format or tags.
- toBlock (string) - The block number as a string in hexadecimal format or tags.
- address (string) - The contract address or a list of addresses from which logs should originate.
- topics (string) - An array of DATA topics and also, the topics are order-dependent.

**Returns**
- 

- result - It returns a filter id to be used when calling eth_getFilterChanges

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
**Description**:
- Adds a new method: `eth_newFilter`, which returns a `filterId` and stores in cache the filter parameters for later usage by other methods.
- Adds a feature flag for all filter API related methods: `FILTER_API_ENABLED`

**Related issue(s)**:

Fixes #1485  

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/relay/src/formatters.ts
Comment (Nana-EC) [inline packages/relay/src/formatters.ts:28]: nit: 
```suggestion
const generateRandomHex = (bytesLength = 16) => {
```

File: packages/relay/src/lib/errors/JsonRpcError.ts
Comment (Nana-EC) [inline packages/relay/src/lib/errors/JsonRpcError.ts:240]: If the range is too large or invalid. Might be good to add a custom message to distinguish when say toBlock is before fromBlock vs when the range is too large

File: packages/relay/src/index.ts
Comment (Nana-EC) [inline packages/relay/src/index.ts:137]: Need to update the API README under `/docs`

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Looking good.
Can you refactor the layout to match the suggested reorg similar to https://github.com/hashgraph/hedera-json-rpc-relay/pull/1590
Left some additional suggestions

File: packages/relay/tests/lib/eth.spec.ts
Comment (Ivo-Yankov) [inline packages/relay/tests/lib/eth.spec.ts:4475]: This tests the case where the block range is too large

File: packages/relay/src/lib/errors/JsonRpcError.ts
Comment (Ivo-Yankov) [reply packages/relay/src/lib/errors/JsonRpcError.ts:240]: This case is already covered and there is a test for it: https://github.com/hashgraph/hedera-json-rpc-relay/pull/1595#discussion_r1284222480

**Comment by Ivo-Yankov** [comment]
> Looking good. Can you refactor the layout to match the suggested reorg similar to #1590 Left some additional suggestions

I'll wait until #1590 is merged and then I'll refactor the layout

**Comment by Ivo-Yankov** [comment]
I refactored the code to follow the new structure and created a new `CommonService` class that will be used for logic that will be shared between other services. Further improvements to the structure, like creating a `BaseService` class, or better separation of the `common` methods can be made in future PRs when we focus on the refactoring of the relay.

**Review by Nana-EC — APPROVED** [review]
LG, let's see if we can note the alpha state of the API

**Pull request merged**
